src/main/java/mahesh/util/TaskList.java

```diff
@@ -0,0 +1,150 @@
+
+/**
+ * Represents a list of tasks.
+ * Provides methods to add, delete, mark, unmark, find, and print tasks.
+ */
+public class TaskList {
+    private ArrayList<Task> list;
+
+    /**
+     * Constructs an empty TaskList.
+     */
+    public TaskList() {
+        this.list = new ArrayList<>();
+    }
+
+    /**
+     * Constructs a TaskList with the specified list of tasks.
+     *
+     * @param list the list of tasks to initialize the TaskList with
+     */
+    public TaskList(ArrayList<Task> list) {
+        this.list = list;
+    }
+
+    /**
+     * Adds a task to the list and increments the task count.
+     *
+     * @param task the task to be added to the list
+     */
+    public void addToList(Task task) {
+        this.list.add(task);
+        Ui.printTaskAdded(task, this.list.size());
+    }
+
+    /**
+     * Deletes a task from the list at the specified index and decrements the task count.
+     *
+     * @param index the index of the task to be removed from the list
+     * @throws IndexOutOfBoundsException if the index is out of range (index < 0 || index >= size())
+     */
+    public void deleteFromList(int index) {
+        try {
+            Task task = this.list.get(index);
+            this.list.remove(index);
+            Ui.printTaskDeleted(task, this.list.size());
+        } catch (IndexOutOfBoundsException err) {
+            Ui.printNoSuchTaskErr(this.list.size());
+        }
+    }
+
+    /**
+     * Marks a task as done at the specified index.
+     *
+     * @param index the index of the task to be marked as done
+     * @throws IndexOutOfBoundsException if the index is out of range (index < 0 || index >= size())
+     */
+    public void markTaskAsDone(int index) {
+        try {
+            Task task = this.list.get(index);
+            task.markAsDone();
+            Ui.printMarkedAsDone(task);
+        } catch (IndexOutOfBoundsException err) {
+            Ui.printNoSuchTaskErr(index);
+        }
+    }
+
+    /**
+     * Unmarks a task as done at the specified index.
+     *
+     * @param index the index of the task to be unmarked as done
+     * @throws IndexOutOfBoundsException if the index is out of range (index < 0 || index >= size())
+     */
+    public void unmarkTaskAsDone(int index) {
+        try {
+            Task task = this.list.get(index);
+            task.unmarkAsDone();
+            Ui.printUnmarkedAsDone(task);
+        } catch (IndexOutOfBoundsException err) {
+            Ui.printNoSuchTaskErr(index);
+        }
+    }
+
+    /**
+     * Finds tasks in the list that match the given search term.
+     * Prints the matching tasks or an error message if no matches are found.
+     *
+     * @param searchTerm the term to search for in the task descriptions
+     */
+    public void findTaskInList(String searchTerm) {
+        if (this.list.size() == 0) {
+            Ui.printEmptyListErr();
+            return;
+        }
+        StringBuilder result = new StringBuilder();
```

Comment:
Small thing, but great use of StringBuilder to improve time complexity even though it might have negligible performance improvements :)